pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadUtils.java

```diff
@@ -107,6 +109,32 @@ public static int sendSegmentFile(final String host, final String port, final St
     return sendFile(host, port, SEGMENTS_PATH, fileName, inputStream, lengthInBytes, SendFileMethod.POST);
   }
 
+
+  public static int sendSegmentUri(final String host, final String port, final String uri) {
+    HttpClient client = new HttpClient();
+    SendFileMethod httpMethod = SendFileMethod.POST;
+    try {
+      client.getParams().setParameter("http.protocol.version", HttpVersion.HTTP_1_1);
+      client.getParams().setSoTimeout(3600 * 1000); // One hour
+      EntityEnclosingMethod method = httpMethod.forUri("http://" + host + ":" + port + "/" + SEGMENTS_PATH);
+      method.setRequestHeader(UPLOAD_TYPE, FileUploadType.Uri.toString());
+      method.setRequestHeader(DOWNLOAD_URI, uri);
```

Comment:
It would be a lot better if we uploaded a JSON object with instructions as opposed to doing an empty post with a HTTP header.